common/ehub_collector.js

```diff
@@ -31,39 +31,31 @@ const defaultProcessError = function(context, err, messages) {
 module.exports = function (context, eventHubMessages, parseFun, processErrorFun, callback) {
     var processError = processErrorFun ? processErrorFun : defaultProcessError;
     var collector = new AlAzureCollector(context, 'ehub', pkg.version);
-    async.reduce(eventHubMessages, {processed: 0, skipped: 0},
-        function(acc, msgArray, callback) {
+    async.reduce(eventHubMessages, [],
+        function(acc, message, reduceCallback) {
+            return reduceCallback(null,[...acc, ...message.records]);
+        },
+        function(err, redResult) {
             try {
-                collector.processLog(msgArray.records, parseFun, null,
+                collector.processLog(redResult, parseFun, null,
                     function(err) {
                         if (err) {
-                            acc.skipped += processError(context, err, msgArray);
+                            const skipped =  processError(context, err, redResult);
+                            context.log.error(`Error while processing records. Skipped ${skipped} Records`);
+                            return callback(err);
```

Comment:
please make sure this error is captured.  Most likely it is output as an invocation result

Reply:
Actually should we return `null, { processed:0, skipped: skipped }` here

Reply:
II was wondering whether to return somthing there myself. Really i dont know if we need to report these numbers because now the whole batch either succeeds or fails.